torchrec/distributed/planner/types.py

```diff
@@ -99,9 +101,15 @@ def __init__(
                 )
             )
 
-        self._local_world_size: int = (
-            local_world_size if local_world_size else world_size
-        )
+        if local_world_size is None:
+            try:
+                local_world_size = int(os.getenv("LOCAL_WORLD_SIZE"))
+            except (TypeError, ValueError):
+                warnings.warn("LOCAL_WORLD_SIZE is not set in Topology. Set it to global WORLD_SIZE. "
+                              "It is valid only when there is only ONE node in the world.")
+                local_world_size = world_size
```

Comment:
also warning when set `local_world_size = world_size`.

It is VERY HARD to debug when local_world_size is incorrect because the previous error call stacks are no related to `Topology`.

The previous call stacks when local_world_size is incorrect are

```
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torchrec/distributed/embedding_lookup.py", line 258, in __init__
    self._lookups.append(sharding.create_lookup(fused_params=fused_params))
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torchrec/distributed/sharding/cw_sharding.py", line 207, in create_lookup
    self._emb_modules.append(_create_lookup(config, device))
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torchrec/distributed/embedding_lookup.py", line 230, in _create_lookup
    return GroupedPooledEmbeddingsLookup(
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torchrec/distributed/embedding_lookup.py", line 258, in __init__
    return BatchedFusedEmbeddingBag(
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torchrec/distributed/batched_embedding_kernel.py", line 641, in __init__
    self._emb_modules.append(_create_lookup(config, device))
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torchrec/distributed/embedding_lookup.py", line 230, in _create_lookup
    self._optim: EmbeddingFusedOptimizer = EmbeddingFusedOptimizer(
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torchrec/distributed/batched_embedding_kernel.py", line 176, in __init__
    return BatchedFusedEmbeddingBag(
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torchrec/distributed/batched_embedding_kernel.py", line 641, in __init__
    local_weight_shards.append(Shard(local_weight, local_metadata))
  File "<string>", line 5, in __init__
    self._optim: EmbeddingFusedOptimizer = EmbeddingFusedOptimizer(
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torchrec/distributed/batched_embedding_kernel.py", line 176, in __init__
    local_weight_shards.append(Shard(local_weight, local_metadata))
  File "<string>", line 5, in __init__
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torch/distributed/_shard/sharded_tensor/shard.py", line 34, in __post_init__
    raise ValueError(
ValueError: Local shard tensor device does not match with local Shard's placement! Found local shard tensor device: cuda:0, local shard metadata placement device: cuda:2
  File "/home/mpi/reyoung_disk/envs/torchrec_dev/lib/python3.9/site-packages/torch/distributed/_shard/sharded_tensor/shard.py", line 34, in __post_init__
    raise ValueError(
ValueError: Local shard tensor device does not match with local Shard's placement! Found local shard tensor device: cuda:0, local shard metadata placement device: cuda:1
[E ProcessGroupNCCL.cpp:416] Some NCCL operations have failed or timed out. Due to the asynchronous nature of CUDA kernels, subsequent GPU operations might run on corrupted/incomplete data. To avoid this inconsistency, we are taking the e
ntire process down.
terminate called after throwing an instance of 'std::runtime_error'
  what():  NCCL error: unhandled system error, NCCL version 2.10.3
ncclSystemError: System call (e.g. socket, malloc) or external library call failed or device error. It can be also caused by unexpected exit of a remote peer, you can check NCCL warnings for failure reason and see if there is connection c
losure by a peer.
```